File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

pair<int, string> removeZero(string s){
    int len = s.size();
    int ret = 0;
    string r = "";
    for(int i = 0; i < len; i ++){
        if(s[i] == '0') ret++;
    }
    int afterlen = len - ret;
    while(afterlen != 0){
        r = to_string(afterlen % 2) + r; 
        afterlen /= 2;
    }
    return {ret, r};
}



vector<int> solution(string s) {
    int a = 0;
    int b = 0;
    pair<int, string> now = {-1, ""};
    while(true){
        now = removeZero(s);
        a++;
        b += now.first;
        s = now.second;
        if(s == "1") break;
        
    }
    
    vector<int> answer = {a, b};
    return answer;
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    cnt = 0
    zero = 0
    while(True):
        if s == '1': break  
        new_s = 0
        for now in s:
            if now == '1': 
                new_s += 1
            else:
                zero += 1
        
        s = bin(new_s)[2:]
        cnt += 1
        
    answer = [cnt, zero]
    
    return answer